map/generate-offline-map-overrides/src/main/java/com/esri/samples/generate_offline_map_overrides/GenerateOfflineMapOverridesController.java

```diff
@@ -131,111 +127,103 @@ private void generateOfflineMap() {
       var offlineMapTask = new OfflineMapTask(map);
 
       // get default offline map parameters for this task given the download area
-      ListenableFuture<GenerateOfflineMapParameters> generateOfflineMapParametersFuture = offlineMapTask
-        .createDefaultGenerateOfflineMapParametersAsync(downloadArea.getGeometry());
+      offlineMapTask.createDefaultGenerateOfflineMapParametersAsync(downloadArea.getGeometry())
+        .toCompletableFuture().whenComplete((parameters, ex) -> {
+          if (ex == null) {
+            // get additional offline parameters (overrides) for this task
+            offlineMapTask.createGenerateOfflineMapParameterOverridesAsync(parameters).toCompletableFuture()
+              .thenAccept(overrides -> {
+                  try {
+                    // get the export tile cache parameters for the base layer
+                    OfflineMapParametersKey basemapParamKey = new OfflineMapParametersKey(
```

Comment:
```suggestion
                    var basemapParamKey = new OfflineMapParametersKey(
```
Where we have `new ClassName` I think we can use more `var`s to just reduce some words.